homeassistant/components/philips_js/binary_sensor.py

```diff
@@ -64,6 +64,11 @@ async def async_setup_entry(
         )
 
 
+def _recordings_list_available(api: PhilipsTV) -> bool:
+    """Return False if recordings_list is None Type."""
+    return not isinstance(api.recordings_list, type(None))
+
+
 def _check_for_recording_entry(api: PhilipsTV, entry: str, value: str) -> bool:
     """Return True if at least one specified value is available within entry of list."""
     for rec in api.recordings_list["recordings"]:
```

Comment:
Just add a check before thks line with return false.

Reply:
if api.recordings_list is None:
  return false

Reply:
You mean instead of:  `return not isinstance(api.recordings_list, type(None))` ?
(I think Github does not indicate the line exactly again)

If I would add it to _check_for_recording_entry, the function will wrongly return false for all and indicate, that "no" recording is available or no recording is running instead of the more correct "unknown".

Reply:
What about this construct?

```
def _check_for_recording_entry(api: PhilipsTV, entry: str, value: str) -> bool | None:
    """Return True if at least one specified value is available within entry of list."""
    if api.recordings_list is None:
        return None

    for rec in api.recordings_list["recordings"]:
        if rec.get(entry) == value:
            return True
    return False
```

Reply:
Just thought about it again.
The three-state idea in my last comment will turn the entities to _unknown_ everytime the TV decides to be in a deep-sleep and outputs None. So it will jump between _unknown_ and _on/off_ in the worst case.

My PRs implementation will just update the state to _on/off_ if there is an update by the TV and just the initial startup-state will be _unknown_.

Reply:
I see little benefit with three state.. Just have it go false.